src/Test/UnitTests/attributes.jl

```diff
@@ -128,3 +128,103 @@ function solve_time(model::MOI.ModelLike, config::TestConfig)
     end
 end
 unittests["solve_time"] = solve_time
+
+# Model and Optimizer attributes.
+for attr in (
+    :BarrierIterations,
+    :ConflictStatus,
+    :DualStatus,
+    :Name,
+    :NodeCount,
+    :NumberOfThreads,
+    :NumberOfVariables,
+    :ObjectiveFunctionType,
+    :ObjectiveSense,
+    :PrimalStatus,
+    :RelativeGap,
+    :ResultCount,
+    :Silent,
+    :SimplexIterations,
+    :SolverName,
+    :SolveTime,
+    :TerminationStatus,
+    :TimeLimitSec,
+)
+    f = Symbol("test_attribute_$(attr)")
+    @eval begin
+        function $(f)(model::MOI.ModelLike, config::TestConfig)
+            MOI.empty!(model)
+            MOI.optimize!(model)
+            attribute = MOI.$(attr)()
```

Comment:
Maybe skip the test is `supports` is false

Reply:
For now I've made these tests opt-in individually (They aren't in the `unittest` testset). We probably want to wait a few releases for solvers to adapt before enforcing it and breaking every solver.

Reply:
What are the changes needed in the solver ? I assume they already have objects of the correct type returned, don't they ?

Reply:
They will return the correct object, but they may not infer correctly.